src/fs.zig

```diff
@@ -159,3 +157,75 @@ pub fn serveFile(
         remaining -= try std.os.sendfile(out, file.handle, len - remaining, remaining, &.{}, &.{}, 0);
     }
 }
+
+test "File server test" {
+    if (@import("builtin").single_threaded) return error.SkipZigTest;
+    const Server = @import("server.zig").Server;
+    const net = std.net;
+
+    const test_alloc = std.testing.allocator;
+    const test_message = "Hello, Apple pie!";
+    const address = try net.Address.parseIp("0.0.0.0", 8080);
+    var server = Server.init();
+
+    const server_thread = struct {
+        var _addr: net.Address = undefined;
+
+        fn index(ctx: void, resp: *Response, req: Request) !void {
+            _ = ctx;
+            try serve({}, resp, req);
+        }
+        fn runServer(context: *Server) !void {
+            try context.run(test_alloc, _addr, {}, index);
+        }
+    };
+    server_thread._addr = address;
+
+    // setup files to serve
+    const cwd = std.fs.cwd();
+    cwd.makeDir("tmp-dir-for-tests") catch |err| switch (err) {
```

Comment:
You can actually use `std.testing.tmpDir` for this. This will make sure we do not dirty the working directory of the user in the event something goes wrong and the delete action didn't work.